### DISCUSSION TITLE
part-ii/2021-27/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#343] [Question comments]
Part II 2021 Automata and Formal Languages: Paper 1, Section II, F 

https://questions.tripos.org/part-ii/2021-27/

**Comment by artie2000** [comment]
<details>
<summary>(i)</summary>

Recursion with base case constant 1 and recursive case constant 0.

</details>
<details>
<summary>(ii)</summary>

Recursion with base case constant 0 and recursive case $s(r(n-1))$.

</details>
<details>
<summary>(iii)</summary>

Minimisation on $r(n)$. Can't be defined without minimisation since it's not total.

</details>
<details>
<summary>What is the class of functions...</summary>

The constant functions, along with $f(a_1,\dots,a_k)=a_j+m$ for fixed $j$ and $m$.

To show these are computable, combine the emptying and transferring subroutines from lectures with a finite number of $(R_1+1)$ states. Special cases are needed for $j=1$ and $m=0$ above.

</details>
